connector/connect/common/src/main/protobuf/spark/connect/relations.proto

```diff
@@ -394,6 +395,18 @@ message CachedLocalRelation {
   string hash = 3;
 }
 
+// Represents a DataFrame that has been cached on server.
+message CachedDataFrame {
+  // (Required) An identifier of the user which cached the dataframe
+  string userId = 1;
```

Comment:
We can also just get userId and sessionId from server via request, instead of passing from here.
But that would require we update [transformRelation()](https://github.com/apache/spark/blob/master/connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala#L87) to take into two more parameters, which means all all those `transform...()` need to be updated to have two more parameter.